### PR TITLE
Fix cql-io bug where restarting C* cluster could cause downtime

### PR DESCRIPTION
**Pull request by akshaymankar**
Upstream changes in https://gitlab.com/twittner/cql-io/-/merge_requests/20

JIRA: https://wearezeta.atlassian.net/browse/SQCORE-1345

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

**Comment by akshaymankar** [comment]
Please approve this PR only if you approve of the upstream MR.

File: cabal.project.freeze
Comment (fisx) [inline cabal.project.freeze:1156]: why is this added here?  don't mind either way, just surprised.

**Review by fisx — APPROVED** [review]

File: cabal.project.freeze
Comment (akshaymankar) [reply cabal.project.freeze:1156]: I just ran the script in `tools/convert-to-cabal` to keep `stack.yaml` and  `cabal.project` in sync. I was confused too, but then thought maybe someone added some dependency and forgot to run that script.

**Review by jschaul — COMMENTED** [review]
Is it possible to create a test for this behaviour before/after; or is that not so easy? If not, would it be possible to detail the manual test procedure steps either here in wire-server in some docs; or in the MR upstream on how you tested the change? (internal jira comments can't be read by cql-io maintainers or anyone else interested)

**Review by jschaul — APPROVED** [review]

**Comment by akshaymankar** [comment]
> Is it possible to create a test for this behaviour before/after; or is that not so easy? If not, would it be possible to detail the manual test procedure steps either here in wire-server in some docs; or in the MR upstream on how you tested the change? (internal jira comments can't be read by cql-io maintainers or anyone else interested)

I think it is quite hard to create an automated test for this since it requires talking to kubernetes. I wrote about my method of testing in the upstream MR: https://gitlab.com/twittner/cql-io/-/merge_requests/20#note_1074472850

**Pull request merged**
